tests/bdd/test_volume_topology.py

```diff
@@ -0,0 +1,555 @@
+        # Since in this scenario, the only pool having topology labels is being used up, we are left with
+        # 0 pools having topology labels.
+        no_of_suitable_pools(
+            volume["spec"]["topology"]["pool_topology"]["labelled"]["inclusion"]
+        )
+        - 1
+        == 0
+    )
+
+
+@when("a user attempts to increase the number of volume replicas")
+def a_user_attempts_to_increase_the_number_of_volume_replicas(replica_ctx):
+    """a user attempts to increase the number of volume replicas."""
+    volumes_api = common.get_volumes_api()
+    volume = volumes_api.get_volume(VOLUME_UUID)
+    num_replicas = volume.spec.num_replicas
+    try:
+        volume = volumes_api.put_volume_replica_count(VOLUME_UUID, num_replicas + 1)
+        replica_ctx[REPLICA_CONTEXT_KEY] = volume.spec.num_replicas
+    except Exception as e:
+        replica_ctx[REPLICA_ERROR] = e
+
+
+@when("the number of suitable pools is less than the number of desired volume replicas")
+def the_number_of_suitable_pools_is_less_than_the_number_of_desired_volume_replicas(
+    create_request,
+):
+    """the number of suitable pools is less than the number of desired volume replicas."""
+    num_volume_replicas = create_request[CREATE_REQUEST_KEY]["replicas"]
+    no_of_pools = no_of_suitable_pools(
+        create_request[CREATE_REQUEST_KEY]["topology"]["pool_topology"]["labelled"][
+            "inclusion"
+        ]
+    )
+    assert num_volume_replicas > no_of_pools
+
+
+@when(
+    "the number of volume replicas is less than or equal to the number of suitable pools"
+)
+def the_number_of_volume_replicas_is_less_than_or_equal_to_the_number_of_suitable_pools(
+    create_request,
+):
+    """the number of volume replicas is less than or equal to the number of suitable pools."""
+    num_volume_replicas = create_request[CREATE_REQUEST_KEY]["replicas"]
+    if "pool_topology" in create_request[CREATE_REQUEST_KEY]["topology"]:
+        no_of_pools = no_of_suitable_pools(
+            create_request[CREATE_REQUEST_KEY]["topology"]["pool_topology"]["labelled"][
+                "inclusion"
+            ]
+        )
+    else:
+        # Here we are fetching all pools and comparing its length, because if we reach this part of code
+        # it signifies the volume request has no pool topology labels, thus all pools are suitable
+        no_of_pools = len(common.get_pools_api().get_pools())
+    assert num_volume_replicas <= no_of_pools
+
+
+@given("additional unused pools with labels containing volume topology")
+def additional_unused_pools_with_labels_containing_volume_topology():
+    """additional unused pools with labels containing volume topology."""
+    volume = common.get_volumes_api().get_volume(VOLUME_UUID)
+    assert (
+        no_of_suitable_pools(
+            volume["spec"]["topology"]["pool_topology"]["labelled"]["inclusion"]
+        )
+        != 0
```

Comment:
Should we check that the number of labelled pools is greater than the current number of volume replicas? That way we know that there are additional pools that can be used. If we just check that the number of suitable pools is not zero, we could potentially already have replicas on them and so won't be able to reuse them.